compiler/cpp/src/thrift/generate/t_go_generator.cc

```diff
@@ -809,17 +812,35 @@ string t_go_generator::render_includes(bool consts) {
     }
 
     string go_module = get_real_go_module(include);
+    string go_path = go_module;
     size_t found = 0;
     for (size_t j = 0; j < go_module.size(); j++) {
       // Import statement uses slashes ('/') in namespace
       if (go_module[j] == '.') {
-        go_module[j] = '/';
+        go_path[j] = '/';
         found = j + 1;
       }
     }
 
-    result += "\t\"" + gen_package_prefix_ + go_module + "\"\n";
-    unused_prot += "var _ = " + go_module.substr(found) + ".GoUnusedProtection__\n";
+    auto it = package_identifiers_.find(go_module);
+    if (it == package_identifiers_.end()) {
+      auto value = go_module.substr(found);
+      // This final path component has already been used, let's construct a more unique alias
+      if (package_identifiers_set_.find(value) != package_identifiers_set_.end()) {
+        // TODO: This would produce more readable code if it appended trailing go_module
+        // path components to generate a more readable name unique identifier (e.g. use
+        // packageacommon as the alias for packagea/common instead of common=). But just
+        // appending an integer is much simpler code
+        value = tmp(value);
+      }
+      package_identifiers_set_.insert(value);
+      it = package_identifiers_.emplace(go_module, std::move(value)).first;
+    }
+    auto const& package_identifier = it->second;
+
+    result += "\t" + package_identifier + " \"" + gen_package_prefix_ +
```

Comment:
Actually it'd be better if I didn't include the alias if it's not necessary. I'll go change that.